pipelines/common/src/main/java/com/google/fhir/analytics/DwhFiles.java

```diff
@@ -192,12 +192,15 @@ public boolean hasIncrementalDir() throws IOException {
    * This method returns the list of non-empty directories which contains at least one file under
    * it.
    *
-   * @param findResourceTypes If true, the views returned will be valid FHIR Resource Types. If false, the
-   *     views will be valid View Definitions
+   * @param findResourceTypes If true, the views returned will be valid FHIR Resource Types. If
+   *     false, the views will be valid View Definitions
+   * @param viewManager Used to verify if non-empty directory is a valid View Definition, Should not
+   *     be set if {@param findResourceTypes} is true
    * @return the set of non-empty directories
    * @throws IOException if the directory does not contain a valid file type
    */
-  public Set<String> findNonEmptyDirs(boolean findResourceTypes) throws IOException {
+  public Set<String> findNonEmptyDirs(boolean findResourceTypes, @Nullable ViewManager viewManager)
```

Comment:
nit: With the new `viewManager` argument, `findResourceTypes` is kind of redundant. Can we change this to `private` and have the following two `public` methods instead:
- `findNonEmptyResourceDirs()` (no arguments)
- `findNonEmptyViewDirs(ViewManager viewManager)` (not-nullable `viewManager` argument).

Reply:
Done.